### DISCUSSION TITLE
How to use terminal escape sequences in prompt?

### DISCUSSION THREAD

**Discussion by riley-van-hengstum** [JanDeDobbeleer/oh-my-posh#3285] [Troubleshoot]
I'm having the following issue. The default cursor shape in my Windows Terminal configuration is a Bar (|) shape. But when a program changes the cursor shape (f.e. vim sets the cursor to a block shape in normal mode)  and then exits back to the prompt, the cursor shape in the terminal doesn't change back to the default cursor shape.

I know I can set the cursor shape back to the default value using an escape sequence. For example when I set the following custom prompt in Powershell:
```powershell
function prompt { "$ `e[0 q" }
```
then the cursor changes back to its default shape every time the prompt displays, and my issue is resolved.

But I'm having a hard time incorporating that solution into oh-my-posh. I can't just add the escape sequence to the template:
```json
"segments": [
  {
    "foreground": "#ffffff",
    "style": "plain",
    "template": "$ `e[0 q",
    "type": "text",
    "interactive": true
  }
]
```
because the escape sequence gets escaped (even when I set `interactive` to `true`).

Is there any way I do this in the oh-my-posh theme? Or are there better ways to do this?

**Comment by riley-van-hengstum** [comment]
Ok, I have something that mostly works.

This code is now in my Powershell profile.
```powershell
oh-my-posh --init --shell pwsh --config ~/scoop/apps/oh-my-posh/current/themes/stelbent.minimal.omp.json | Invoke-Expression

# get script block for oh-my-posh prompt
$prompt = (Get-Command Prompt).ScriptBlock

function prompt { 
    # set cursor to default shape
    Write-Host -NoNewLine "`e[0 q"
    # execute script block for oh-my-posh prompt
    Invoke-Command $prompt 
}
```

For some reason I don't understand yet, if I reload the profile (`&$profile`) then it reverts back to the old prompt. But other than that it seems to work.

**Comment by JanDeDobbeleer** [comment]
@Jaap-van-Hengstum as this is called from go, you can't really use a PowerShell only reference to `ESC`, you can try this instead `\x1b[0 q`. Let me know if that works.
